src/tdastro/astro_utils/passbands.py

```diff
@@ -102,134 +168,149 @@ class Passband:
 
     Attributes
     ----------
-    label : str
-        The label of the passband. Note that this is not unique across surveys.
     survey : str
-        The survey to which the passband belongs.
+        The survey to which the passband belongs: eg, "LSST".
+    filter_name : str
+        The name of the passband's filter: eg, "u".
     full_name : str
-        The full name of the passband.
+        The full name of the passband. This is the survey and filter concatenated: eg, "LSST_u".
     table_path : str
         The path to the transmission table file.
     table_url : str
         The URL to download the transmission table file.
-    transmission_table : np.ndarray
+    waves : np.ndarray
+        The wavelengths of the transmission table. To be used when evaluating models to generate fluxes
+        that will be passed to fluxes_to_bandflux.
+    _loaded_table : np.ndarray
+        A 2D array of wavelengths and transmissions. This is the table loaded from the file, and is neither
+        interpolated nor normalized.
+    processed_transmission_table : np.ndarray
         A 2D array where the first col is wavelengths (Angstrom) and the second col is transmission values.
-    normalized_transmission_table : np.ndarray
-        A 2D array of wavelengths and normalized transmissions.
+        This table is both interpolated to the _wave_grid and normalized to calculate phi_b(λ).
     """
 
     def __init__(
         self,
         survey: str,
-        label: str,
+        filter_name: str,
+        delta_wave: Optional[float] = 5.0,
+        trim_quantile: Optional[float] = 1e-3,
         table_path: Optional[str] = None,
         table_url: Optional[str] = None,
         units: Optional[Literal["nm", "A"]] = "A",
+        force_download: bool = False,
     ):
-        """Construct a Passband object
+        """Construct a Passband object.
 
         Parameters
         ----------
         survey : str
-            Which survey does this passband correspond to. Often "LSST"
-        label : str
-            Which label does this passband correspond to, Sometimes a filter name like "g"
+            The survey to which the passband belongs: eg, "LSST".
+        filter_name : str
+            The filter_name of the passband: eg, "u".
+        delta_wave : float or None, optional
+            The grid step of the wave grid, in angstroms.
+            It is typically used to downsample transmission using linear interpolation.
+            Default is 5 angstroms. If `None` the original grid is used.
+        trim_quantile : float or None, optional
+            The quantile to trim the transmission table by. For example, if trim_quantile is 1e-3, the
+            transmission table will be trimmed to include only the central 99.8% of the area under the
+            transmission curve.
         table_path : str, optional
-            Path to the table defining the passband on the filesystem. Will take precedence over table_url
```

Comment:
I think it is fine to keep this as-is, but an alternative approach that might be more modular approach might be to have the constructor take a table and then to add two class methods:

```
@classmethod
def from_file(cls, table_path: str, ...):
   load the file
   ...
   return Passband(table, ...)

@classmethod
def from_url(cls, table_url: str, force_download: bool, ...):
    do the download
    ...
    return Passband(table, ...)
```

Reply:
Referenced in #113